returnn/datasets/distrib_files.py

```diff
@@ -413,6 +456,59 @@ def _get_key_for_file_tree(t: FileTree) -> str:
     return ":".join(tree.flatten(t))
 
 
+def _get_rank_and_size() -> Tuple[int, int]:
+    """
+    gets the global rank and size for multiprocess trainings
+    """
+
+    from returnn.config import get_global_config
+
+    config = get_global_config(raise_exception=False)
+    if not config:
+        return 0, 1
+    if config.typed_value("torch_distributed") is not None:
+        import returnn.torch.distributed
+
+        ctx = returnn.torch.distributed.get_ctx(config=config)
+        return ctx.rank(), ctx.size()
+    elif config.is_true("use_horovod"):
+        assert config.bool("use_tensorflow", False) or config.value("backend", "").startswith("tensorflow")
+
+        import returnn.tf.horovod
+
+        ctx = returnn.tf.horovod.get_ctx(config=config)
+        return ctx.rank(), ctx.size()
+    else:
+        return 0, 1
+
+
+def _parse_num_shards_input(num_shards_input: Union[None, int, str]) -> Tuple[int, int]:
+    """
+    parses the given input for the number of shards into an integer
+    and returns the local shard index for this worker
+    :param num_shards_input: the user-configured value for the number of shards
```

Comment:
```suggestion
    :param num_shards_input: the user-configured value for the number of shards
    :return: Local shard index for this worker and global number of shards.
```
Maybe also delete the line above, `and returns...`?

Reply:
No, there should always be a `:return:` section in the docstring.

Reply:
Yes, that's why I asked this. If I remember correctly there wasn't a `:return:` section in the string, as I proposed in the suggestion above. My proposal here was removing

> and returns the local shard index for this worker

and adding it more explicitly as a `:return:` tag.